byway/src/main/webapp/destinations.js

```diff
@@ -218,21 +237,69 @@ function getCurrentAddress(){
 * add event listener for submit button
 */
 window.onload = function(){
+  
   let nextButton = document.getElementById('next-button');
   nextButton.addEventListener('click', () => {
-      nextButton.href = buildUrlWithParams("/interests.html", {tripKey});
+      nextButton.href = configureTripKeyForPath(tripKey, '/interests.html');
   });
   document.getElementById('user-input-form').addEventListener('submit', (event) => {
     event.preventDefault();
-    const formData = new  FormData(document.getElementById("user-input-form"));
-    fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
-        response.json()).then(locationData => {
-        updateLocations(locationData);
-        updateStartDestination(locationData);
-      });
-    });
+    savePlaceIds();
+  });
 }
 
+/** 
+* gets User Input from search box and sends the Place Id of user input to the server
+* elementName indicates which search box to get from, param indicates which param to save as
+* @param {String} elementName
+*/
+function savePlaceIds(){
+  let formData = new FormData();
+
+  const destRequest = {
+    query: String(document.getElementById("destinations-search-box").value),
+    fields: ["place_id"]
+  };
+  const startRequest = {
+    query: String(document.getElementById("start-search-box").value),
+    fields: ["place_id"]
+  };
+
+  let promise = new Promise((resolve,reject) => {
+    placesService.findPlaceFromQuery(destRequest, (results, status) => {
+    console.log(status);
+      if (status === google.maps.places.PlacesServiceStatus.OK) {
+        formData.append("destinations-search-box", results[0].place_id);
+        resolve(results);
+      }
+      else{
+        alert("Status: " + status);
+        reject(new Error("Could not find place:" + status));
+      }
+    });
+  }).catch(error => {
+    alert("Error: cannot process this request due to " + error);
+  });
```

Comment:
When structuring code like this, where you bridge a callback-based API into a `Promise`, it's better to keep the initial promise as narrowly-scoped as possible. Don't do anything in the initial `new Promise(...)` except call the callback-based API and resolve/reject with the result, then handle any extra/external actions by chaining off of that promise.

In fact, when you have a common API which you need to bridge into a promise, I would recommend making a separate function which returns a promise-based version of the same work as the callback-based code, something like:

```js
function findPlaceFromQuery(request) {
  return new Promise(resolve => {
    placesService.findPlaceFromQuery(request, (result, status) => resolve({result, status}));
  });
}
```

That way you have a simple function which you can use to make an async query, like this:

```js
findPlaceFromQuery(destRequest)
  .then(({result, status}) => {
    // ...
  });
```

(Note that in this code, I've twice used `{result, status}`. In `resolve({result, status})` I am packing both the result and status into an object together, since `resolve` can only take one value and I want to pass both, the only way to do that is to put both `result` and `status` in an object together. In the second example block where I write `({result, status}) => { ... }`, I'm doing the reverse: I'm *unpacking* `result` and `status` from that object to get them both back as separate variables in the "then" handler.)

Reply:
ok added this function